drivers/audio/codec_shell.c

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2023 Centralp
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/shell/shell.h>
+#include <zephyr/audio/codec.h>
+
+#define CODEC_START_HELP                                                                           \
+	"Start output audio playback. Syntax:\n"                                                   \
+	"<device>"
+
+#define CODEC_STOP_HELP                                                                            \
+	"Stop output audio playback. Syntax:\n"                                                    \
+	"<device>"
```

Comment:
it'd be interesting to clarify what 'start/stop' mean in the context of an audio codec
a) start the interface, e.g. I2S
b) power-up the analog parts
c) all of the above
d) all of the above + more?

Also why is this limited to playback?

Reply:
> it'd be interesting to clarify what 'start/stop' mean in the context of an audio codec a) start the interface, e.g. I2S b) power-up the analog parts c) all of the above d) all of the above + more?

I use the same description as the one in [Audio Codec documentation](https://docs.zephyrproject.org/latest/hardware/peripherals/audio/codec.html#c.audio_codec_start_output).

Stopping audio codec is setting codec output to mute or high impedance.
Starting audio codec is setting codec output to unmute or play.

That not affects I2S part (which is the input of the audio codec).

Maybe I will propose an `audio` command later to manage the I2S part (play/stop audio data through I2S.

Reply:
thanks for the pointer, I wasn't aware there was an audio codec API. My comments are really at the API level, the same opens on what 'start' or 'stop' means, muting or high-impedance are radically different.

But if this PR is just adding a shell to an existing API I have no objection.